src/qtism/runtime/tests/RouteItem.php

```diff
@@ -450,4 +450,68 @@ public function getTimeLimits($excludeItem = false): RouteTimeLimitsCollection
 
         return $timeLimits;
     }
+
+    public function getEffectiveBranchRules(): BranchRuleCollection
+    {
+        if ($this->getBranchRules()->count() > 0) {
+            return $this->getBranchRules();
+        }
+
+        $sectionBranchRules = $this->getEffectiveSectionBranchRules();
+
+        if ($sectionBranchRules === null || $sectionBranchRules->count() > 0) {
+            return $sectionBranchRules ?? new BranchRuleCollection();
+        }
+
+        $testPartSections = $this->getTestPart()->getAssessmentSections()->getArrayCopy();
+
+        if (
+            end($testPartSections) === $this->getAssessmentSection()
+            && $this->getTestPart()->getBranchRules()->count() > 0
+        ) {
+            return $this->getTestPart()->getBranchRules();
+        }
+
+        return new BranchRuleCollection();
+    }
+
+    /**
+     * Selects branching rules from the section/subsection.
+     * Branching rules will be selected only if the item or subsection is the last one in the parent section.
+     *
+     * @return ?BranchRuleCollection Returns the branching rules for the last section or null if the element/subsection
+     *                               is not the last.
+     */
+    public function getEffectiveSectionBranchRules(): ?BranchRuleCollection
+    {
+        $sections = $this->getAssessmentSections()->getArrayCopy();
+        $currentSection = array_pop($sections);
```

Comment:
Isn't this the same?

```suggestion
        $currentSection = $this->getAssessmentSection();
```

Reply:
The same. But `array_pop` will help me later when I'm reaching `foreach` statement.
Just to explain:
* `->getSectionParts()` on current section (means, subsection or single section) will return list of items for this section.
* `->getSectionParts()` on parent section (means, parent of subsection) will return list of subsections

So when I'm doing `array_pop` - I'm receiving current section. And I'm doing necessary checks for this.
And then, I'm iterating through the list of parent sections and checking them.
So, here `array_pop` helps me to avoid iterating through a single section (which will return list of items, but I need list of parent sections).

Sorry for difficult explanation 🤯